packages/fuels-core/src/types/bits_256.rs

```diff
@@ -102,4 +163,37 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_param_type_evm_addr() {
+        assert_eq!(
+            EvmAddress::param_type(),
+            ParamType::Struct {
+                fields: vec![ParamType::B256],
+                generics: vec![]
+            }
+        );
+    }
+
+    #[test]
+    fn test_from_token_evm_addr() -> Result<(), Error> {
+        let data = [0u8; 32];
+        let token = Token::Struct(vec![Token::B256(data)]);
+
+        let evm_address = EvmAddress::from_token(token)?;
+
+        assert_eq!(evm_address.value.0, data);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_into_token_evm_addr() {
+        let data = [0u8; 32];
+        let evm_address = EvmAddress::from(Bits256(data));
+
+        let token = evm_address.into_token();
+
+        assert_eq!(token, Token::Struct(vec![Token::B256(data)]));
```

Comment:
I think it would be better t o check that the data is valid on non-0 vectors, here you're not making sure that it works as it should with the conversion I think